tests/acceptance/test_serializers.py

```diff
@@ -141,6 +143,28 @@ async def test_multi_set_multi_get_types(self, cache, obj):
         assert await cache.multi_get([Keys.KEY]) == [pickle.loads(pickle.dumps(obj))]
 
 
+class TestMsgPackSerializer:
+    TYPES = (1, 2.0, "hi", True, ["1", 1], {"key": "value"})
+
+    @pytest.mark.parametrize("obj", TYPES)
+    async def test_set_get_types(self, cache, obj):
+        cache.serializer = MsgPackSerializer()
+        assert await cache.set(Keys.KEY, obj) is True
+        assert await cache.get(Keys.KEY) == msgpack.loads(msgpack.dumps(obj))
```

Comment:
I think some of the failures are related to the test itself (specifically `msgpack.loads(msgpack.dumps(obj))`)?
https://github.com/aio-libs/aiocache/actions/runs/8513779973/job/23320587633?pr=836#step:7:82

Reply:
I don't think the test itself is causing issues here; adding

```py
assert msgpack.loads(msgpack.dumps(obj)) == obj
```

right before the other asserts in `TestMsgPackSerializer` doesn't raise assertion errors.

(I just noticed that some tests fail with a `TypeError: a bytes-like object is required, not 'str'` error though, I wonder what's causing that.)

Reply:
> My guess would be that `TypeError: a bytes-like object is required, not 'str'` is raised because of the same encoding issues as the `UnicodeDecodeError` ones. But it's not raised by msgpack itself, that's for sure.

That's the one I was linking to. ;)